mathics_scanner/tokeniser.py

```diff
@@ -259,7 +259,8 @@
     "^": ["UpSetDelayed", "UpSet", "Power"],
     "_": ["Pattern"],
     "`": ["Pattern", "Symbol"],
-    "|": ["RawRightAssociation", "Or", "Alternatives"],
+    # FIXME: the -> part of |-> in function isn't handled
```

Comment:
So we should increase the precendence of `|->`.

Reply:
The precedence of `|` 160, so if the precedence of `|->` be greater than that this won't occurs anymore, I suppose.

Reply:
Yes, that works (mathics-core tests passes). I set it 800 in the YAML table. If you want I can push that.

Reply:
Great! Please push on top.

Reply:
Done.